tests/docker-images/java-test-functions/src/main/java/org/apache/pulsar/tests/integration/io/TestGenericObjectSink.java

```diff
@@ -56,10 +56,11 @@ public void write(Record<GenericObject> record) {
 
         if (record.getSchema().getSchemaInfo().getType() == SchemaType.KEY_VALUE) {
             // assert that we are able to access the schema (leads to ClassCastException if there is a problem)
-            KeyValueSchema kvSchema = (KeyValueSchema) record.getSchema();
-            log.info("key schema type {}", kvSchema.getKeySchema());
-            log.info("value schema type {}", kvSchema.getValueSchema());
-            log.info("key encoding {}", kvSchema.getKeyValueEncodingType());
+            // TODO need to expose KeyValueSchema was an interface in pulsar-client-api
+//            KeyValueSchema kvSchema = (KeyValueSchema) record.getSchema();
```

Comment:
This is a braking change for the new feature.

Please do not change this behaviour now

Reply:
It's not a breaking change because `KeyValueSchema` is not part of public API. Only symbols that are in `pulsar-client-api` are. I think the test here should use `Schema<KeyValue>` instead.